.github/workflows/deploy.yml

```
@@ -0,0 +1,77 @@
+  push:
+    branches: [ main ]
+
+env:
+  DOCKER_IMAGE: your-dockerhub-username/hono-api
+  DOCKER_TAG: ${{ github.sha }}
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)